components/AnimationTiming/examples/supplemental/AnimationTimingExampleSupplemental.m

```diff
@@ -54,14 +54,54 @@ - (void)setupExampleViews {
   self.title = @"Animation Timing";
 
   self.scrollView = [[UIScrollView alloc] initWithFrame:self.view.bounds];
-  self.scrollView.autoresizingMask =
-      UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
 
   self.scrollView.contentSize = CGSizeMake(CGRectGetWidth(self.view.frame),
                                            CGRectGetHeight(self.view.frame) + kTopMargin);
   self.scrollView.clipsToBounds = YES;
   [self.view addSubview:self.scrollView];
 
+#if defined(__IPHONE_11_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_11_0)
+  if (@available(iOS 11.0, *)) {
+    // No need to do anything - additionalSafeAreaInsets will inset our content.
+    self.scrollView.autoresizingMask =
+        UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
+  } else {
+#endif
+    self.scrollView.translatesAutoresizingMaskIntoConstraints = NO;
+    [NSLayoutConstraint activateConstraints:
+     @[[NSLayoutConstraint constraintWithItem:self.scrollView
+                                    attribute:NSLayoutAttributeTop
+                                    relatedBy:NSLayoutRelationEqual
+                                       toItem:self.topLayoutGuide
+                                    attribute:NSLayoutAttributeBottom
+                                   multiplier:1.0
+                                     constant:0],
+       [NSLayoutConstraint constraintWithItem:self.scrollView
+                                    attribute:NSLayoutAttributeBottom
+                                    relatedBy:NSLayoutRelationEqual
+                                       toItem:self.view
+                                    attribute:NSLayoutAttributeBottom
+                                   multiplier:1.0
+                                     constant:0],
+       [NSLayoutConstraint constraintWithItem:self.scrollView
+                                    attribute:NSLayoutAttributeLeft
+                                    relatedBy:NSLayoutRelationEqual
+                                       toItem:self.view
+                                    attribute:NSLayoutAttributeLeft
+                                   multiplier:1.0
+                                     constant:0],
+       [NSLayoutConstraint constraintWithItem:self.scrollView
+                                    attribute:NSLayoutAttributeRight
+                                    relatedBy:NSLayoutRelationEqual
+                                       toItem:self.view
+                                    attribute:NSLayoutAttributeRight
+                                   multiplier:1.0
+                                     constant:0]
```

Comment:
Is there a more concise way for me to say "attached to left/right/bottom edges and the top layout guide?"

Reply:
there are layout anchors, but they are only iOS9+

Reply:
Sad trombone.